gratipay/models/package/__init__.py

```diff
@@ -40,3 +51,35 @@ def from_names(cls, package_manager, name):
         """
         return cls.db.one("SELECT packages.*::packages FROM packages "
                           "WHERE package_manager=%s and name=%s", (package_manager, name))
+
+
+    @property
+    def team(self):
+        """A computed attribute, the :py:class:`~gratipay.models.team.Team`
+        linked to this package if there is one, otherwise ``None``. Makes a
+        database call.
+        """
+        return self._load_team(self.db)
+
+
+    def ensure_team(self, cursor, owner):
+        """Given a db cursor and :py:class:`Participant`, insert into ``teams`` if need be.
+        """
+        team = self._load_team(cursor)
```

Comment:
Ah! I think it's because the ... well, wait ...

I was gonna say because the calls synchronize on the `cursor.run('DELETE ...')` call (why that one instead of the `SELECT`s? Is psycopg2/Postgres being fancy?), so that the second cursor sees a certain state following the lock release, ... but wouldn't a second insert on `teams_to_packages` still fail?